### ISSUE TITLE
Decrease in working set was incorrectly marked as test failure

### ISSUE THREAD

**Issue by jeremymeng** [dotnet/corefx#323]
I am hitting a test failure that does not have a consistent repro.
```
                     Err_004888anied Total memory increased significantly there may be a memory leak (TaskId:339)
                     Percent Difference=0.338227637196709, Max Acceptable Percent Difference=0.3, Start Working Set=576180, End Working Set=381300 (TaskId:339)
```

We probably should not use `Math.abs()` to calculate the difference at https://github.com/dotnet/corefx/blob/master/src/System.Text.RegularExpressions/tests/RegexLangElementsCoverageTests.cs#L60

```csharp
            percentDifference = ((double)Math.Abs(startWorkingSet - endWorkingSet)) / startWorkingSet;

            if (maxAcceptablePercentDifference < percentDifference)
            {
                Console.WriteLine("Err_004888anied Total memory increased significantly there may be a memory leak");
                Console.WriteLine("Percent Difference={0}, Max Acceptable Percent Difference={1}, Start Working Set={2}, End Working Set={3}",
                    percentDifference, maxAcceptablePercentDifference, startWorkingSet, endWorkingSet);
                iCountErrors++;
            }
```

**Comment by mellinoe** [comment]
Sorry, my mind had slipped about this over the end of the holidays. I'm of the opinion that we should probably just remove this check entirely, but if it is the case that failures are just occurring when memory usage DECREASES, then it would probably be a simple fix to just calculate the actual difference.

**Comment by jeremymeng** [comment]
I agree that we should remove the memory usage checks.  Or add a dedicate stress test if we care about this.

**Comment by sharwell** [comment]
This test will obviously not work as expected if tests are being run in parallel.

This *type* of test can be objectively implemented using snapshots in the memory profiler. This is the manner in which I deterministically evaluate the memory behavior of ANTLR 4 (in that case, I can examine object reachability from a specific set of locations to prevent delayed collection of objects from skewing the result).

**Comment by stephentoub** [comment]
Fixed by #408
